### DISCUSSION TITLE
Is Supabase support unicode characters?

### DISCUSSION THREAD

**Discussion by onurozkan** [supabase/supabase#408] [Questions] [answered]
I created a table on app.supabase.io, added existed content in new table screen, but all turkish characters (çÇşŞöÖüÜ) corrupted and turned into Ã¼l like characters.

Couldnt find any documentation/option on website about changing collation.

So, is there any UTF-8 support?

**Comment by inian** [comment]
Hi @onurozkan ,the default encoding is already set to UTF8. I just created a row with "çÇşŞöÖüÜ" from the Supabase dashboard and it is inserted correctly. 

Could you share more details on how you are inserting your records and a sample record which is not getting inserted correctly?

**Reply by soedirgo** [reply]
I tried this with supabase-js, it does seem like there's a problem in handling Unicode:
<img width="871" alt="Screenshot 2020-12-18 at 1 26 40 PM" src="https://user-images.githubusercontent.com/31685197/102578297-cab69400-4134-11eb-9a59-c56aa420ca2f.png">
The one above is inserted using the table editor, the one below with supabase-js.

I'll create an issue for this in postgrest-js. Thanks for reporting @onurozkan!

**Reply by onurozkan** [reply]
@soedirgo @inian 

Actually I imported content through new table => add existing content (CSV )option.

![Screen Shot 2020-12-18 at 09 17 51](https://user-images.githubusercontent.com/20716/102581620-fd02ca00-4111-11eb-9e50-966536d65415.png)
![Screen Shot 2020-12-18 at 09 18 02](https://user-images.githubusercontent.com/20716/102581623-fe33f700-4111-11eb-8810-52284fee7792.png)

If you add a record through "add new row" option, it works okay.

**Form**
![Screen Shot 2020-12-18 at 09 13 03](https://user-images.githubusercontent.com/20716/102581437-a1384100-4111-11eb-91e8-f190768098d0.png)

**Result**
![Screen Shot 2020-12-18 at 09 14 36](https://user-images.githubusercontent.com/20716/102581383-7e0d9180-4111-11eb-864e-9c22be719b27.png)

**Comment by joshenlim** [comment, marked as answer]
Hey @onurozkan! If it's convenient will it be alright if you were to share with us the CSV you used here?

**Reply by onurozkan** [reply]
Hi @joshenlim 

Original one contains private information, so I created a new csv with lots of turkish characters, so you can test the behavior. 

I couldn't attach CSV to github, here is the content of CSV.

```csv
"Id","Title","FirstName","LastName","FullName","UserName","Picture"
"646","Developer","Ahmet","Yıldırım","Ahmet Yıldırım","ayildirim","ahmet_yildirim_F0005.jpg"
"554","Developer","Ayhan","İskandinavoğlu","Ayhan İskandinavoğlu","aiskandinavoglu","aiskandinavoglu.jpg"
"275","Developer","Aslıhan","Kemerci","Aslıhan Kemerci","akemerci","asl_han.jpg"
"348","Developer","Aygül","Keçi","Aygül Keçi","akeci","aygul.jpg"
"395","Developer","Ayşegül","Cankuş","Ayşegül Cankuş","acankus","aysegul_5F0A9.jpg"
"630","Developer","Şemsettin","Erdinç","Şemsettin Erdinç","serdinc","sems_erd.jpg"
"664","Developer","Fıskiye","Tılı","Fıskiye Tılı","ftili","fiskiye.jpg"
"653","Developer","Çaçan","Burhan","Burhan Çaçan","bcacan","burcu_bilge_89ED5.jpg"
"260","Developer","Belkıs","Odabaşı","Belkıs Odabaşı","bodabasi","belkis_odabasi-copywriting.jpg"
"462","Web Developer","Buğra","Demir","Buğra Demir","bdemir","bugra_1.jpg"
"655","Developer","Büşra","Şemensi","Büşra Şemensi","bsemensi","busra_ru_2B2CB.jpg"

```

**Reply by joshenlim** [reply]
This is perfect @onurozkan thanks so much! We've managed to find where it went wrong for the CSV import - a fix will be pushed out shortly :)

**Reply by inian** [reply]
Hi @onurozkan the fix for this has been deployed. Can you please help test again? Thanks!

**Reply by onurozkan** [reply]
@inian @joshenlim @soedirgo 

Sorry guys, I missed the github notification. I can confirm it is working as expected.

Thank you

**Reply by Aspendas** [reply]
Hi @inian I am having a similar issue. When I import CSV to superbase, some characters from Turkish, like ı, turn into �.

**Reply by sabimasar** [reply]
@joshenlim hi! I am also having issues with the characters not showing properly. This is for the special characters across all languages such as è, ß etc. My csv is UTF-8 and shows properly but when uploaded to Supabase using the CSV import, the characters are replaced with �.

**Reply by joshenlim** [reply]
hey there! if it's convenient would you mind sharing what your CSV looks like and we can help to further investigate thereafter 🙂🙏 Also some screenshots around the issue will help a lot too!

**Reply by sabimasar** [reply]
sure, here 
<img width="205" alt="Screenshot 2025-04-04 at 15 48 33" src="https://github.com/user-attachments/assets/226857de-0e04-4e97-b379-69300e524b83" />
[
[sample.csv](https://github.com/user-attachments/files/19605514/sample.csv)
](url)
<img width="380" alt="Screenshot 2025-04-04 at 15 47 17" src="https://github.com/user-attachments/assets/4a824c99-ab2f-4653-b34a-2f592e9ed1ec" />

**Reply by joshenlim** [reply]
@sabinabale  Okay I think I see what's happening here - I'm guessing you're using Microsoft Excel to export your CSV? It seems like Microsoft Excel doesn't export CSVs into utf-8 format ([reference](https://github.com/mholt/PapaParse/issues/169#issuecomment-73717741)) (Or the export isn't in the right format at least) Had a hunch this might be the case too as the characters didn't show up properly when I opened your CSV in VScode too

Based on the above attached comment, I tried to upload your CSV to Google Sheets and then re-exported it and can verify that the import shows the contents correctly 🙂 🙏 

[Untitled spreadsheet - sample.csv](https://github.com/user-attachments/files/19606488/Untitled.spreadsheet.-.sample.csv)

![image](https://github.com/user-attachments/assets/dc8e2dd9-6eea-4e98-bfa7-7a5c4dfa6aaf)

**Reply by sabimasar** [reply]
Actually, I went back and tried to upload the original csv I had and it works. The issue was me trying to convert it to make the file "readable" to me but I should have left it alone. I assumed that if I can't read it, it won't show the way I need it to in Supabase, but I was wrong. Computer understands computer. So sorry to have wasted your time with this and thank you! 🙏🏻

original csv (looks incorrect but it's fine):
<img width="224" alt="Screenshot 2025-04-04 at 17 32 20" src="https://github.com/user-attachments/assets/450930a7-e468-460c-a9c3-29cff1fa5ca4" />

in supabase (shows correctly):
<img width="413" alt="Screenshot 2025-04-04 at 17 32 37" src="https://github.com/user-attachments/assets/fd1bef22-df92-419b-bfe7-2a70617c46a0" />

PS: Simply uploading the already converted csv file to Google Drive and downloading it does not work and the screenshot you shared doesn't show the correct characters still. But leaving the original csv file alone works for me.

The letters should look like this:
<img width="372" alt="Screenshot 2025-04-04 at 17 41 16" src="https://github.com/user-attachments/assets/c9c0e17d-45e9-4752-a811-d6ccbafc515c" />

**Reply by joshenlim** [reply]
Oh not at all! Glad to hear it all worked out for you at the end 😄🙏 Happy to help! 🙂
